gbpservice/neutron/nsf/configurator/api/v1/controllers/controller.py

```diff
@@ -0,0 +1,181 @@
+
+    def _update_network_service_config(self, body):
+        request_data = body.get("request_data")
+        try:
+            return json.dumps(
+                self.rpcclient.update_network_service_config(request_data))
+        except Exception as e:
+            return json.dumps({'err_msg': e.args})
+
+    def _delete_network_device_config(self, body):
+        request_data = body.get("request_data")
+        try:
+            return json.dumps(
+                self.rpcclient.delete_network_device_config(request_data))
+        except Exception as e:
+            return json.dumps({'err_msg': e.args})
+
+    def _delete_network_service_config(self, body):
+        request_data = body.get("request_data")
+        try:
+            return json.dumps(
+                self.rpcclient.delete_network_service_config(request_data))
+        except Exception as e:
+            return json.dumps({'err_msg': e.args})
+
+
+class RPCClient(object):
+
+    """send RPC call/cast on behalf of controller class
+    according to the curl request"""
+    API_VERSION = '1.0'
+
+    def __init__(self, topic, host):
+
+        self.topic = topic
+        self.host = host
+        target = oslo_messaging.Target(
+            topic=self.topic,
+            version=self.API_VERSION)
+        n_rpc.init(cfg.CONF)
+        self.client = n_rpc.get_client(target)
+
+    def get_notifications(self):
+
+        cctxt = self.client.prepare(server=self.host)
+        return cctxt.call(self, 'get_notifications')
+
+    def create_network_device_config(self, request_data):
+
+        cctxt = self.client.prepare(server=self.host)
+        return cctxt.cast(self, 'create_network_device_config',
+                          request_data=request_data)
+
+    def create_network_service_config(self, request_data):
+
+        cctxt = self.client.prepare(server=self.host)
+        return cctxt.cast(self, 'create_network_service_config',
+                          request_data=request_data)
+
+    def update_network_device_config(self, request_data):
+
+        cctxt = self.client.prepare(server=self.host)
+        return cctxt.cast(self, 'update_network_device_config',
+                          request_data=request_data)
+
+    def update_network_service_config(self, request_data):
+
+        cctxt = self.client.prepare(server=self.host)
+        return cctxt.cast(self, 'update_network_service_config',
+                          request_data=request_data)
+
+    def delete_network_device_config(self, request_data):
+
+        cctxt = self.client.prepare(server=self.host)
+        return cctxt.cast(self, 'delete_network_device_config',
+                          request_data=request_data)
+
+    def delete_network_service_config(self, request_data):
+
+        cctxt = self.client.prepare(server=self.host)
+        return cctxt.cast(self, 'delete_network_service_config',
+                          request_data=request_data)
+
+    def to_dict(self):
```

Comment:
Why is this needed? We should avoid this method. Context class must be having this.